src/components/va-back-to-top/va-back-to-top.tsx

```diff
@@ -0,0 +1,78 @@
+import { Component, Element, Host, State, h } from '@stencil/core';
+import classnames from 'classnames';
+
+import 'intersection-observer';
+
+@Component({
+  tag: 'va-back-to-top',
+  styleUrl: 'va-back-to-top.css',
+  shadow: true,
+})
+export class VaBackToTop {
+  revealPixel!: HTMLSpanElement;
+
+  @State() hasHitBreakpoint = false;
+  @State() isDocked = false;
+
+  @Element() el: HTMLElement;
+
+  componentDidLoad() {
+    // Setup observer to handle docking behavior
+    const dockObserver = new IntersectionObserver(
+      entries => {
+        // The second half of the || ensures that the button
+        // stays docked even if the viewport is scrolled below the dock
+        this.isDocked =
+          entries[0].isIntersecting || entries[0].boundingClientRect.y < 0;
+      },
+      { threshold: 1.0 },
+    );
+    dockObserver.observe(this.el);
+
+    // Setup observer to handle reveal behavior
+    const revealObserver = new IntersectionObserver(entries => {
+      this.hasHitBreakpoint = entries[0].boundingClientRect.y < 0;
+    });
+    revealObserver.observe(this.revealPixel);
+  }
+
+  navigateToTop() {
+    // Focus the h1 tag on the page.
+    const el = document.querySelector('h1');
+    if (el) {
+      // Prepare the element so that it can accept focus properly.
+      el.setAttribute('tabindex', '-1');
+      el.focus();
+
+      // Cleanup the tabindex on blur.
+      el.addEventListener('blur', () => el.removeAttribute('tabindex'));
+    }
+
+    // Scroll to the top.
+    return window.scrollTo(0, 0);
+  }
+
+  render() {
+    const undockedWidth = this.el.getBoundingClientRect().width;
+    this.el.style.setProperty('--undocked-width', `${undockedWidth}px`);
+
+    return (
+      <Host>
+        <span
+          class="reveal-point"
+          ref={el => (this.revealPixel = el as HTMLSpanElement)}
+        ></span>
```

Comment:
When the viewport has scrolled enough to where this DOM node is above it, the button is revealed. This is different from the [current implementation](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/6fc5f410037d867f998ace61a1398c6b802cea00/packages/formation/js/back-to-top.js#L37-L38) which compares `window.scrollY` against a magic number of 600.

Reply:
I recommend documenting how this component works. Probably in a JSDoc on the `@Component()`. It's clever, which means folks who aren't privy to how it works right now might struggle figuring it out later.